Realm/RLMRealm.mm

```diff
@@ -71,8 +71,7 @@ - (void)notify;
 
 // create NSException from c++ exception
 static __attribute__((noreturn)) void throw_objc_exception(exception &ex) {
-    NSString *errorMessage = [NSString stringWithUTF8String:ex.what()];
-    @throw [NSException exceptionWithName:@"RLMException" reason:errorMessage userInfo:nil];
+    @throw RLMException(ex);
 }
 
 // create NSError from c++ exception
```

Comment:
Should convert make_realm_error to RLMError following the pattern used with RLMException.